### DISCUSSION TITLE
Correct way of importing an RDF Knowledge Graph into PyG

### DISCUSSION THREAD

**Discussion by igrangel** [pyg-team/pytorch_geometric#2857] [Q&A]
Hi!

I have an RDF/OWL based KG which have some classes, object properties, etc. I want to use it in PyG. 

1.  I have read it with rdflib from its turtle serialization
```
rg = RDFGraph()
rg.parse("data.ttl", format='turtle')
print("rdflib Graph loaded successfully with {} triples".format(len(rg)))
```
2. Then I have transformed it into the NetworkX library
```
G = rdflib_to_networkx_graph(rg)
print("networkx Graph loaded successfully with length {}".format(len(G)))
```
3. And finally, I used  the method from_network to bring it into the PyG data. 

```
from torch_geometric.utils import from_networkx
pG = from_networkx(G)
```

Is this the correct way to bring these types of KGs into PyG?

**Comment by rusty1s** [comment]
In theory, yes. Although you probably want to group all your different attributes into single feature matrices afterwards. Let me know if you need any further help :)

**Reply by igrangel** [reply]
Thanks a lot for your reply. Would be great to have some example, do you happen to have something like this?

What do you mean by 

> Although you probably want to group all your different attributes into single feature matrices afterwards

If you can be more specific, I would really appreciate it ;).

**Reply by rusty1s** [reply]
I mean that, depending on how your input graph looks like, you will have many different feature vectors in the resulting `data` object. You can simply print it out to confirm. When working with GNNs, you will probably want to group them into a single feature matrix of shape `[num_nodes, num_total_features]`.
